docs/UpgradeGuide.md

```diff
@@ -2,57 +2,52 @@
 
 # From 0.x to 1.0
 
-- [Namespacing & the public API](#namespacing--the-public-api)
-- [Configuration](#configuration)
-- [Instrumentation](#instrumentation)
-- [Full list of breaking changes](#full-list-of-breaking-changes)
-
 Upgrading `ddtrace` from 0.x to 1.x introduces some changes to namespacing, the public API, and the underlying trace data structure.
 
-Here's a list of the most common changes you may encounter:
-
-## Namespacing & the public API
-
-To avoid naming conflicts with new Datadog features and products, many of the tracing constants and functions defined directly under `Datadog` have moved to `Datadog::Tracing`.
-
-You can access most common functions through our [public API](https://www.rubydoc.info/gems/ddtrace/) within `Datadog::Tracing`. Use of some of the functions in the public API will be described in use cases below. Here's a few examples:
-
-```ruby
-### Old 0.x ###
-Datadog.tracer.trace
-Datadog.tracer.active_span
-Datadog.tracer.active_correlation.to_s
-
+**How to upgrade basic usage** *(Out-of-the-box instrumentation)*
 
-### New 1.0 ###
-Datadog::Tracing.trace
-Datadog::Tracing.active_span
-Datadog::Tracing.log_correlation
-# ...and more...
-```
+- [Configuration](#1.0-configuration)
+  - [Settings have been renamed](#1.0-configuration-settings)
+  - [Activating instrumentation](#1.0-configuration-instrumentation)
+- [Instrumentation](#1.0-instrumentation)
+  - [Service naming](#1.0-instrumentation-service-naming)
 
-Check out our [complete list of namespacing changes](#appendix-namespacing) for more details.
+**Addtional upgrades for advanced usage** *(Manual instrumentation)*
 
-### Removed `Datadog.tracer`
+- [Namespacing](#1.0-namespacing)
+- [Trace API](#1.0-trace-api)
+  - [Removed `Datadog.tracer`](#1.0-trace-api-removed-tracer)
+  - [Removed access to `Datadog::Context`](#1.0-trace-api-removed-context)
+  - [Manual tracing & trace model](#1.0-trace-api-manual-tracing)
+  - [Accessing trace state](#1.0-trace-api-trace-state)
+  - [Distributed tracing](#1.0-trace-api-distributed)
+    - [Over HTTP](#1.0-trace-api-distributed-http)
+    - [Over gRPC](#1.0-trace-api-distributed-grpc)
+    - [Between threads](#1.0-trace-api-distributed-threads)
+  - [Sampling](#1.0-trace-api-sampling)
+  - [Processing pipeline](#1.0-trace-api-pipeline)
 
-Many of the functions accessed directly through `Datadog.tracer` have been moved to `Datadog::Tracing` instead.
+**Appendix**
 
-### Removed access to `Datadog::Context`
+  - [Namespace mappings](#1.0-appendix-namespace)
+    - [`require` paths](#1.0-appendix-namespace-requires)
+    - [Constants](#1.0-appendix-namespace-constants)
+  - [Breaking changes](#1.0-appendix-breaking-changes)
 
-Direct usage of `Datadog::Context` has been removed. Previously, it was used to modify or access active trace state. Most use cases have been replaced by our public trace API.
+<h1 id="1.0-basic-upgrade">Upgrading basic usage</h1>
 
```

Comment:
I touched a bit on this on my suggestions to the intro, but I think we should clarify at the beginning of this section what we mean by basic[/standard] usage, because while it's quite clear to us the developers which one is which, customers will probably not know on which bucket they are.

Here's my suggestion for that intro. It repeats a bit what I suggested we say above, but  makes more explicit what we mean by "standard" usage.

> For most users, moving to 1.0 is as simple as changing a few (possibly no lines of code).
>
> If you're only using automatic instrumentation and don't use `Datadog.configure` to change the gem, then you're done, no changes are needed! 🎉 
>
> If you're configuring `ddtrace` via `Datadog.configure`, read on for a few changes that you'll need to do.
> 
> If, additionally, you're calling directly into `ddtrace` APIs from your code -- anything beyond `Datadog.configure`, you should additionally look to the _Additional upgrades for advanced usage_ section.

Perhaps we could even add:

> Did we miss something? Please do [get in contact](https://github.com/DataDog/dd-trace-rb/issues) -- we want to ensure your upgrade experience is brilliant!

Reply:
I agree it would be better to clarify what qualifies a user for each group. However, I think this description belongs at the top, and (again) I don't think we should use such a personal or casual tone.